data/data-pipeline/data_pipeline/etl/score/etl_score.py

```diff
@@ -310,61 +316,21 @@ def _add_percentiles_to_df(
 
         else:
             # For agricultural loss, we are using whether there is value at all to determine percentile and then
-            # filling places where the value is False with 0
+            # filling places where the value is False with np.NaN (since it doesn't apply)
+            # For linguistic isolation, we drop PR and then recalculate. We fill PR with np.NaN
             df[
                 f"{output_column_name_root}"
                 f"{field_names.PERCENTILE_FIELD_SUFFIX}"
-            ] = (
-                df.where(
-                    df[field_names.AGRICULTURAL_VALUE_BOOL_FIELD].astype(float)
-                    == 1.0
-                )[input_column_name]
-                .rank(ascending=ascending, pct=True)
-                .fillna(
-                    df[field_names.AGRICULTURAL_VALUE_BOOL_FIELD].astype(float)
-                )
-            )
-
-        # Create the urban/rural percentiles.
-        urban_rural_percentile_fields_to_combine = []
-        for (urban_or_rural_string, urban_heuristic_bool) in [
-            ("urban", True),
-            ("rural", False),
-        ]:
-            # Create a field with only those values
-            this_category_only_value_field = (
-                f"{input_column_name} (value {urban_or_rural_string} only)"
-            )
-            df[this_category_only_value_field] = np.where(
-                df[field_names.URBAN_HEURISTIC_FIELD] == urban_heuristic_bool,
-                df[input_column_name],
-                None,
-            )
-
-            # Calculate the percentile for only this category
-            this_category_only_percentile_field = (
-                f"{output_column_name_root} "
-                f"(percentile {urban_or_rural_string} only)"
-            )
-            df[this_category_only_percentile_field] = df[
-                this_category_only_value_field
+            ] = df.groupby(df[field_names.GEOID_TRACT_FIELD].isin(drop_tracts))[
```

Comment:
Can you add a comment explaining this code? Maybe I'm just not fully awake this morning but I have honestly no idea what's going on in this code. 

Here's my cro-magnon attempt to understand what's happening: 

1. You're limiting the data frame to only include tracts that you're intentionally trying to drop? Why, wouldn't it be the other way? 

2. Then you're running a groupby -- I guess grouping by each of the tracts that are in `drop_tracts`? Why? 

3. Then you're only selecting one column (the input), makes sense
4. Running a percentile on this 
5. Setting percentile to the output column name / percentile field. 

I'm honestly just very confused. It feels like maybe you're using the groupby as a "shortcut" to do something that's not traditionally an aggregation. But this code feels kind of error prone because you're packing so much into a one liner.

Reply:
Going with option 1

Reply:
Updated!